### PR TITLE
`Upload Questions` view

### LINKED ISSUE #35: Frontend: Design a view for an `Upload Question`
Must be, as much as possible, independant from any parent component.

> Priority: __HIGH__

### PR DESCRIPTION
**Pull request by YacineSteeve**
Resolves #35

### CODE REVIEW

File: frontend/src/components/questions/UploadQuestion.vue [hunk 1]
```diff
@@ -1,5 +1,226 @@
+<script setup lang="ts">
+import { computed, ref } from 'vue';
+import type { ComputedRef, Ref } from 'vue';
+import type { Quizoot } from '@interfaces/quizoot';
+
+interface Specs {
```
Comment (chrichri17) [inline frontend/src/components/questions/UploadQuestion.vue]: We are the one who usually use long names.. Except when you really need it 😂 😂 

`UploadQuestionProps` might be better here.

File: frontend/src/components/questions/UploadQuestion.vue [hunk 2]
```diff
@@ -1,5 +1,226 @@
...
+}
+
+function onDragLeave(event: DragEvent) {
+    event.preventDefault();
+    isDragOver.value = false;
+}
+
+function onFileAdd(event: HTMLInputEvent | DragEvent) {
+    event.preventDefault();
+    const files =
```
Comment (chrichri17) [inline frontend/src/components/questions/UploadQuestion.vue]: You might want to implement a type guard here.

```typescript
const isDragEvent(event: HTMLInputEvent | DragEvent): event is DragEvent => 'dataTransfer' in event;

const getFiles = (event: HTMLInputEvent | DragEvent) => {
    if (isDragEvent(event)) {
        return event.dataTransfer?.files;
    }
    return event.target.files;
}

**Review by chrichri17 — APPROVED** [review]
Looks good. I have to test it as well.

File: frontend/src/components/questions/UploadQuestion.vue (hunk 1 above)
Comment (YacineSteeve) [reply frontend/src/components/questions/UploadQuestion.vue]: Argh, poor me 😭

**Pull request merged**